File: safe-online-sdk-dotnet/src/main/cs/ws/notification/NotificationProducerClient.cs
/*
 * SafeOnline project.
 * 
 * Copyright 2006-2008 	Lin.k N.V. All rights reserved.
 * Lin.k N.V. proprietary/confidential. Use is subject to license terms.
 */

using System;

namespace safe_online_sdk_dotnet
{
	public interface NotificationProducerClient
	{
		/// <summary>
		/// Subscription for the application at the specified address to the specified topic.
		/// </summary>
		/// <param name="topic"></param>
		/// <param name="address"></param>
		/// <exception cref="SubscriptionFailedException"></exception>
		/// <exception cref="RuntimeException"></exception>
		void subscribe(string topic, string address);
	}
}

File: safe-online-sdk-api/src/main/java/net/link/safeonline/sdk/api/ws/notification/producer/client/NotificationProducerClient.java
/*
 * SafeOnline project.
 *
 * Copyright 2006-2007 Lin.k N.V. All rights reserved.
 * Lin.k N.V. proprietary/confidential. Use is subject to license terms.
 */

package net.link.safeonline.sdk.api.ws.notification.producer.client;

import net.link.safeonline.sdk.api.exception.SubscriptionFailedException;
import net.link.safeonline.sdk.api.exception.WSClientTransportException;


/**
 * WS Notification Producer Service Client interface.
 *
 * @author wvdhaute
 */
public interface NotificationProducerClient {

    /**
     * Subscribe the specified consumer address to the specified Topic.
     *
     * @throws SubscriptionFailedException
     * @throws WSClientTransportException
     */
    void subscribe(String topic, String address)
            throws SubscriptionFailedException, WSClientTransportException;
}
